packages/data-explorer/src/components/DataRowEdit.vue

```diff
@@ -0,0 +1,266 @@
+        this.isSaving = true
+        try {
+          const response = await repository.save(this.formData, this.formFields, this.entity, this.dataRowId)
+          if (this.parent) {
+            const newOptionLocation = response.headers.get('Location')
+            const createdRow = await repository.fetchOption(newOptionLocation)
+            // Create a new option object to pass to the reference select
+            this.parent.optionCreatedCallback({
+              id: createdRow[this.idAttribute],
+              value: createdRow[this.idAttribute],
+              label: createdRow[this.labelAttribute]
+            })
+            this.showParent()
+          } else {
+            this.goBackToPluginCaller()
+          }
+        } catch (e) {
+          this.handleError(e)
+        }
+      } else {
+        this.saveFailed = true
+      }
+    },
+    onCancelClick () {
+      this.parent ? this.showParent() : this.goBackToPluginCaller()
+    },
+    goBackToPluginCaller () {
+      window.history.go(-1)
+    },
+    clearAlert () {
+      this.alert = null
+    },
+    setRef (ref) {
+      this.showRef = ref
+    },
+    showParent () {
+      this.parent.setRef(false) // show parent
+      this.parent.$refs.refContainer.removeChild(this.$el) // destroy child
+    },
+    handleError (error) {
+      const alertMsg = this.errorToMessage(error)
+      this.showForm = true
+      this.isSaving = false
+      this.alert = { message: alertMsg, type: 'danger' }
+    },
+    errorToMessage (error) {
+      if (typeof error === 'string') {
+        return error
+      } else if (error && Array.isArray(error.errors) && error.errors.length) {
+        return `${error.errors[0].message} (${error.errors[0].code})`
+      } else {
+        return this.$t('data-row-edit-default-error-message')
+      }
+    },
+    /**
+       * Takes molgenis api-v2 metaData object and builds a map from fieldName to referenceEntity name
+       * Only field that have a reference entity are included in the map
+       */
+    buildReferenceMap (metaData) {
+      // recursily walk compound
+      const flattenAttr = (attr) => attr.attributes && attr.attributes.length ? attr.attributes.flatMap(flattenAttr) : [attr]
+
+      return metaData.attributes
+        .flatMap(flattenAttr)
+        .filter(attr => Object.prototype.hasOwnProperty.call(attr, 'refEntity'))
+        .reduce((accum, attr) => {
+          accum[attr.name] = attr.refEntity.name
+          return accum
+        }, {})
+    },
+    async fetchTableData (entity, dataRowId) {
+      const mapperOptions = {
+        showNonVisibleAttributes: true,
+        mapperMode: dataRowId ? 'UPDATE' : 'CREATE',
+        booleanLabels: {
+          trueLabel: this.$t('data-row-edit-boolean-true'),
+          falseLabel: this.$t('data-row-edit-boolean-false'),
+          nillLabel: this.$t('data-row-edit-boolean-null')
+        }
+      }
+      try {
+        const resp = await repository.fetch(entity, dataRowId)
```

Comment:
should probably be moved to store to be consistent with data explorer architecture

Reply:
Refactoring the DataRowEdit component is out of scope for this ticket.